Realm/RLMObjectStore.mm

```diff
@@ -256,9 +209,71 @@ void RLMRealmCreateTables(RLMRealm *realm, RLMSchema *targetSchema, bool updateE
         RLMObjectSchema *tableSchema = [RLMObjectSchema schemaFromTableForClassName:objectSchema.className realm:realm];
         RLMVerifyAndAlignColumns(tableSchema, objectSchema);
     }
+
+    return changed;
+}
+
+NSError *RLMUpdateRealmToSchemaVersion(RLMRealm *realm, NSUInteger newVersion, RLMSchema *targetSchema, NSError *(^migrationBlock)()) {
+    [realm beginWriteTransaction];
+
+    @try {
+        NSUInteger oldVersion = RLMRealmSchemaVersion(realm);
+
+        // validate versions
+        if (oldVersion > newVersion && oldVersion != RLMNotVersioned) {
+            @throw [NSException exceptionWithName:@"RLMException"
+                                           reason:@"Version of Realm file on disk is higher than current schema version"
+                                         userInfo:@{@"path" : realm.path}];
+        }
+
+        // create tables
+        bool migrating = oldVersion != newVersion;
+        bool changed = RLMRealmCreateTables(realm, targetSchema, migrating);
+
+        // apply migration block if provided
+        if (migrating && migrationBlock) {
+            NSError *error = migrationBlock();
+            if (error) {
+                [realm cancelWriteTransaction];
+                return error;
+            }
+        }
+
+        if (oldVersion != newVersion) {
+            RLMRealmSetSchemaVersion(realm, newVersion);
+            changed = true;
+        }
+
+        if (changed) {
+            [realm commitWriteTransaction];
+        }
+        else {
+            [realm cancelWriteTransaction];
+        }
+    }
+    @catch (NSException *) {
+        [realm cancelWriteTransaction];
+        @throw;
+    }
+
+    return nil;
 }
 
 
+void RLMRealmSetSchema(RLMRealm *realm, RLMSchema *targetSchema, bool verify) {
```

Comment:
Can you move this up so it isn't changed in the diff since there are no code changes?